File: backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema(
    {
        productName: {
            type: String,
            required: true
        },
        productDescription: {
            type: String,
            required: true
        },
        productImage: {
            type: String,
            required: true
        },
        stock: {
            type: Number,
            required: true
        },
        price: {
            type: Number,
            required: true,
        }
    }
);

module.exports = mongoose.model('Product', productSchema);

File: frontend/models/product.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Product struct {
	ID                 primitive.ObjectID `bson:"_id,omitempty"`
	ProductName        string             `bson:"productName"`
	ProductDescription string             `bson:"productDescription"`
	ProductImage       string             `bson:"productImage"`
	Stock              int32              `bson:"stock"`
	Price              float32            `bson:"price"`
}
